### PR TITLE
feat: add ACM certificate expiry triage

### LINKED ISSUE #283: feat: add ACM certificate browser with expiry triage
## Summary
Expiring certificates are scheduled outages. unic should surface them before DNS does.

## Scope
- Certificate list sorted by days-to-expiry, with status, in-use-by count, and renewal eligibility.
- Detail: domains/SANs, validation method/status, in-use-by ARNs.
- Security Inspector rule: warn when a certificate expires within N days (default 30).

## Acceptance Criteria
- Soon-to-expire certs surface first with clear day counts.
- Inspector emits expiry findings through the existing rule-pack pattern.
- Tests cover mapping, sorting, and the rule. README/docs updated.

### PR DESCRIPTION
**Pull request by YoungJinJung**
### Summary

- add an expiry-sorted ACM certificate browser with status, renewal, validation, SAN, and usage details
- add a Security Inspector finding for certificates expiring within 30 days
- add mock-backed mapping/sorting and inspector-rule coverage

### Related Issues

Closes #283

### Validation

- `make test`
- `make build`

### Checklist

- [x] Scope is focused
- [x] Branch name follows `docs/branch-naming-harness.md`
- [x] Documentation harness reviewed (`docs/documentation-harness.md`)
- [x] README updated if user-facing behavior changed
- [x] Relevant `docs/` pages updated if architecture, auth, config, or workflow changed
- [x] Tests/validation included
- [x] Breaking changes documented

## Summary by CodeRabbit

* **New Features**
  * Added an ACM Certificate Browser with certificate status, domains, expiry, renewal eligibility, validation details, and resource usage.
  * Added sorting, filtering, refreshing, scrolling, detailed views, and keyboard navigation.
  * Added security findings for expiring certificates, with configurable warning windows and higher severity for certificates expiring within seven days.
* **Documentation**
  * Documented ACM browser features, certificate information, security findings, and keyboard actions.
* **Maintenance**
  * Updated AWS and supporting dependencies.

### CODE REVIEW

**Review by YoungJinJung — COMMENTED** [review]
Reviewed head: 85f0f9006718f575618bc275eab8469ebcb26e07

## Findings

1. **[P2] Render missing expiry values as unavailable** — `internal/app/screen_acm.go:136-137`

   `NotAfter` is optional in ACM certificate details (for example, a certificate that is still pending validation can have no expiry yet). The list correctly renders this case as `-`, but the detail view formats the zero `time.Time` and reports it as `0001-01-01 00:00:00` with `0` days left. That misleadingly makes a non-issued certificate look like it expires immediately. Guard `c.NotAfter.IsZero()` here and render both fields as `-` (or another explicit unavailable value); add a view test covering a certificate with no `NotAfter`.

**Comment by YoungJinJung** [comment]
Addressed the missing-expiry review finding in 78693ab: certificates without `NotAfter` now render `Expires` and `Days Left` as `-`, with a focused detail-view regression test. Validation passed: `make test`, `make build`, and `git diff --check`.

**Review by YoungJinJung — COMMENTED** [review]
Reviewed head: 78693ab4362cd937466a56aeb4f918bf72db57da

## Findings

1. **[P2] Make the variable-length certificate detail scrollable** — `internal/app/screen_acm.go:157`

   The detail view appends every SAN, validation record, and attached resource to one unbounded string, while `HandleKey` for `screenACMCertificateDetail` only handles back navigation. ACM certificates can contain many SANs (and therefore many validation rows) plus multiple `InUseBy` entries, so on a normal-height terminal the later rows and help bar fall below the viewport with no way to reach them. This makes part of the advertised validation and usage detail inaccessible. Add detail scroll state and window the rendered lines using the repository's `visibleLines := max(m.height-N, 5)` pattern, handle navigation/page keys, and cover a certificate whose detail exceeds the viewport.

**Comment by YoungJinJung** [comment]
Addressed the detail-scrolling review finding in e7d82ed: ACM certificate details now use a height-bounded window with line and page navigation, reset scroll state when entering/leaving, and include a regression test that pages to later usage rows. Validation passed: `make test`, `make build`, and `git diff --check`.

File: internal/app/screen_acm.go [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+	b.WriteString(titleStyle.Render("Certificate Detail"))
+	b.WriteString("\n\n")
+	lines := am.detailLines(m)
+	visibleLines := max(m.height-8, 5)
+	start := min(am.detailScroll, max(len(lines)-visibleLines, 0))
+	for _, line := range lines[start:min(start+visibleLines, len(lines))] {
+		b.WriteString(line)
+	}
+	b.WriteString("\n")
+	b.WriteString(m.renderHelpBar("↑/↓: scroll • pgup/pgdn: page • esc: back • H: home"))
```
[the remark this answers is not in the document]
Comment (YoungJinJung) [reply internal/app/screen_acm.go]: Fixed in 2886f36: the detail screen now declares scroll/page bindings in screenKeymaps and renders the shared keymap help.

File: internal/inspector/inspector_rules_acm.go [hunk 2]
```diff
@@ -0,0 +1,45 @@
+package inspector
+
+import (
+	"context"
+	"fmt"
+	"time"
+)
+
+const inspectorACMExpiryDays = 30
```
[the remark this answers is not in the document]
Comment (YoungJinJung) [reply internal/inspector/inspector_rules_acm.go:9]: Not changing this: issue #283 specifies the 30-day expiry policy, and the existing inspector rules use fixed rule thresholds. Adding configuration would expand scope without a stated user requirement.

File: internal/services/aws/acm_test.go [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+		t.Fatal(err)
+	}
+	if len(certificates) != 2 || certificates[0].ARN != "soon" {
+		t.Fatalf("expected soonest expiry first, got %+v", certificates)
+	}
+	soon := certificates[0]
+	if soon.DaysToExpiry(now) != 5 || len(soon.InUseBy) != 1 || len(soon.Validation) != 1 || soon.Region != "us-east-1" {
+		t.Fatalf("unexpected mapping: %+v", soon)
+	}
+}
```
[the remark this answers is not in the document]
Comment (YoungJinJung) [reply internal/services/aws/acm_test.go:69]: Not expanding this into an exhaustive matrix: the added tests cover the new mapping/sorting and expiry severity behavior; list/describe errors are directly wrapped and returned with no additional branching. Focused boundary cases can be added if a concrete defect appears.

File: README.md [hunk 4]
```diff
@@ -443,6 +444,7 @@ checks:
 | SQS | `A` toggle all-regions scope, `/` filter, `r` refresh, `Enter` detail, detail `d` jump to DLQ, `m` redrive DLQ (type-to-confirm), `x` purge (type-to-confirm) |
 | ELB | `A` toggle all-regions scope, `/` filter, `r` refresh, `Enter` target groups, target group list `Enter` per-target health |
 | Parameter Store | `/` filter, `r` refresh, `Enter` detail, detail `v` reveal value (decrypts SecureString), `y` copy value without revealing |
+| ACM Certificates | `/` filter, `r` refresh, `Enter` certificate detail |
```
[the remark this answers is not in the document]
Comment (YoungJinJung) [reply README.md]: Fixed in 2886f36: README now lists ACM in shared filter coverage and documents detail scrolling/page keys.

**Review by YoungJinJung — COMMENTED** [review]
Reviewed head: 2886f3649c4db0e4e2f4bf3610b2cc092533a242

## Finding

1. **[P2] Make the ACM expiry window configurable** — `internal/inspector/inspector_rules_acm.go:9`

   Issue #283 requires findings within “N days (default 30),” but this implementation fixes the policy at `inspectorACMExpiryDays = 30` with no configuration path. That satisfies the default but not the configurable `N`, so users cannot apply a different certificate-renewal policy. Add an inspector/config setting that defaults to 30, pass it into the ACM scan, and cover both the default and an override in tests; document the setting alongside the ACM inspector behavior.

**Comment by YoungJinJung** [comment]
Addressed the configurable-expiry-window review finding in 6596622: `inspector.acm_expiry_window_days` now defaults to 30, supports a positive-day override from `config.yaml`, and is passed into the ACM Security Inspector scan without global mutable state. Added focused default/override coverage and documented the setting. Validation passed: `make test`, `make build`, and `git diff --check`.

**Comment by YoungJinJung** [comment]
Addressed the ACM detail-loading performance finding in d8f186d: certificate details now load concurrently in deterministic batches capped at 10 requests, with regression coverage verifying the bound. Validation passed: `make test`, `make build`, and `git diff --check`.

**Review by YoungJinJung — COMMENTED** [review]
Reviewed head: d8f186d3bec98598a2d6c9b4767def22507a2ef0

No actionable findings. I verified the complete diff, prior review threads and follow-up commits, passing CI, and local `make test` / `make build`. The PR is mergeable but remains blocked on the required independent approval.

File: internal/services/aws/acm.go [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+				wg.Add(1)
+				go func() {
+					defer wg.Done()
+					detailOut, err := r.ACMClient.DescribeCertificate(ctx, &acm.DescribeCertificateInput{CertificateArn: summary.CertificateArn})
+					if err != nil {
+						errs[i] = fmt.Errorf("failed to describe ACM certificate %s: %w", derefString(summary.CertificateArn), err)
+						return
+					}
+					results[i] = detailOut.Certificate
+				}()
```
[the remark this answers is not in the document]
Comment (YoungJinJung) [reply internal/services/aws/acm.go:97]: This is not a race on the current branch. `go.mod` declares Go 1.25, and since Go 1.22 each range iteration creates new `i` and `summary` variables, so each closure captures its own values. The indexed `results`/`errs` writes are also disjoint and read only after `wg.Wait()`. I additionally ran the focused concurrent ACM tests under `-race` 20 times; they passed. No code change is needed.

**Review by YoungJinJung — COMMENTED** [review]
Reviewed head: d8f186d3bec98598a2d6c9b4767def22507a2ef0

No actionable findings. I read the complete current diff, tests, prior review threads, and the latest bounded-concurrency commit. The range-variable race report does not apply under this repository’s Go 1.25 toolchain; batch result writes are disjoint and consumed only after `Wait`.

Validation: `make test`, `make build`, and `git diff --check origin/main...origin/feature/issue-283-acm-browser` all pass. CI is green. I am not self-approving or merging; branch protection still requires an independent approving review.

**Review by YoungJinJung — COMMENTED** [review]
Reviewed head: d8f186d3bec98598a2d6c9b4767def22507a2ef0

## Finding

1. **[P1] Include ACME-origin certificates in the browser and inspector** — `internal/services/aws/acm.go:68`

`ListCertificates` leaves `CertificateKeyPairOrigins` unset. AWS now excludes `ACME` certificates by default, returning only `AWS_MANAGED` and `CUSTOMER_PROVIDED`, so both the certificate browser and expiry inspector silently miss every ACME-origin certificate. Pass all three supported origins (`AWS_MANAGED`, `ACME`, `CUSTOMER_PROVIDED`) in the request and extend the request-shape test to cover them.

**Comment by YoungJinJung** [comment]
Addressed the ACME-origin review finding in b18a892: ACM list requests now include AWS_MANAGED, ACME, and CUSTOMER_PROVIDED certificate origins, with the existing request-shape test covering all three. Validation passed: `make test`, `make build`, and `git diff --check`.

**Comment by YoungJinJung** [comment]
Addressed the bounded-concurrency test finding in d85afcc: the test now uses a one-second context deadline, and blocked mock describe calls return on cancellation instead of hanging indefinitely. Validation passed: `make test`, `make build`, and `git diff --check`.

**Review by YoungJinJung — COMMENTED** [review]
Reviewed head: d85afcceff15646aec6ff63469b40e996ff45932

No actionable findings. I read the complete current diff, tests, CI, prior reviews/comments, and the latest bounded-concurrency commit. The reported goroutine closure race does not apply: this module declares Go 1.25, where range iteration variables are distinct per iteration; each goroutine writes to a disjoint result index, and the slices are consumed only after Wait returns.

Validation: `go test -race ./internal/services/aws`, `make test`, `make build`, and `git diff --check origin/main...HEAD` all pass. CI is green. I am not self-approving or merging; branch protection still requires an independent approving review.

**Review by YoungJinJung — COMMENTED** [review]
Reviewed head: d85afcceff15646aec6ff63469b40e996ff45932

No actionable findings. I read the complete diff, tests, CI, prior reviews, comments, and current review threads. The remaining Amazon Q loop-variable warning is a false positive for this module: `go.mod` declares Go 1.25, where range variables are per-iteration, and writes target distinct slice indexes before `wg.Wait()` returns. Local validation passed: `make test`, `make build`, `git diff --check`, and `go test -race ./internal/services/aws -run TestListCertificates -count=20`.

**Review by YoungJinJung — COMMENTED** [review]
Reviewed head: d85afcceff15646aec6ff63469b40e996ff45932

No actionable findings. The bounded-concurrency regression test now has a one-second context deadline and cancellation-aware mock calls, eliminating the prior indefinite-hang risk. I also verified the complete PR diff and resolved/outdated thread state. Local `make test`, `make build`, and `git diff --check origin/main...HEAD` pass; required CI and Amazon Q checks are green.

**Review by YoungJinJung — COMMENTED** [review]
Reviewed the complete diff at d85afcceff15646aec6ff63469b40e996ff45932, including the final fixes, tests, CI, prior reviews, and comments. No actionable concerns found. Local validation passed: `make test`, `make build`, and `git diff --check`. Approval is not possible from YoungJinJung because this PR is authored by the same GitHub identity.

File: internal/services/aws/acm.go [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
```
[the remark this answers is not in the document]
Comment (YoungJinJung) [reply internal/services/aws/acm.go:97]: This is not a race on the current toolchain. The module targets Go 1.25, and since Go 1.22 each iteration of a `range` loop creates fresh iteration variables, so `i` and `summary` are distinct for every goroutine. The per-index writes to `results` and `errs` are therefore independent; no code change is needed.

File: internal/inspector/inspector_rules_acm.go [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+		if days <= 7 {
+			severity = RuleSeverityHigh
+		}
+		findings = append(findings, SecurityFinding{
+			RuleID:         "acm-certificate-expiry",
+			RuleName:       "ACM certificate expiring soon",
+			Severity:       severity,
+			ResourceType:   "ACM Certificate",
+			ResourceID:     certificate.ARN,
+			Summary:        fmt.Sprintf("Certificate for %s expires in %d days.", certificate.DomainName, days),
```
Comment (YoungJinJung) [inline internal/inspector/inspector_rules_acm.go]: [P2] `DaysToExpiry`가 음수를 0으로 clamp하므로 이미 만료된 인증서도 여기서 `expires in 0 days`로 보고됩니다. 만료 예정과 이미 만료는 대응 우선순위와 운영 의미가 다른데 현재 finding이 이를 숨깁니다. `NotAfter.Before(now)`를 먼저 분기해 `expired N days ago`(또는 최소 `expired`)로 표시하고, 만료 전 0일과 과거 만료를 각각 테스트해 주세요.

**Review by YoungJinJung — COMMENTED** [review]
현재 head 재검토에서 기존 리뷰와 겹치지 않는 만료 상태 표현 문제를 확인했습니다.

File: internal/inspector/inspector_rules_acm.go (hunk 7 above)
Comment (YoungJinJung) [reply internal/inspector/inspector_rules_acm.go]: Fixed in acbed79. Expired certificates now report `has expired`, while the zero-day non-past boundary remains `expires in 0 days`; focused regression coverage verifies both cases. Validation passed: `go test ./internal/inspector`, `make test`, `make build`, and `git diff --check`.

**Review by YoungJinJung — COMMENTED** [review]
Reviewed head: acbed79574ecbec9fa4d3316267d163739663529

## Findings

1. **[P3] Update the required overview and architecture documentation** — `internal/domain/catalog.go:154`, `internal/inspector/inspector_rules_acm.go:12`, `internal/config/config.go:31`

   This change adds a supported AWS service/browser, a new built-in Security Inspector rule, a new configuration field, and a new TUI screen family, but the PR changes no file under `docs/`. The repository's `docs/documentation-harness.md` explicitly maps a new AWS service to `docs/project-overview.*` and `docs/architecture.*`, and requires architecture docs for TUI-flow and config changes. As a result, the English and Korean project overview/architecture pages remain stale about supported service areas, inspector coverage, configuration, and screen families even though the PR checklist says the relevant pages were updated. Update the corresponding `docs/project-overview.en.md`, `docs/project-overview.ko.md`, `docs/architecture.en.md`, and `docs/architecture.ko.md` sections to describe ACM browsing and expiry inspection (including the configurable window and new screens).

**Comment by YoungJinJung** [comment]
Addressed the canonical documentation review finding in 4b9e9d8: the English and Korean project overviews and architecture docs now cover ACM service scope, repository integration, certificate list/detail screens, the expiry Inspector rule, and the configurable warning window. Validation passed: `make test`, `make build`, and `git diff --check`.

File: internal/services/aws/acm_test.go [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+	var active, peak atomic.Int32
+	release := make(chan struct{})
+	mock := &mockACMClient{
+		list: func(context.Context, *acm.ListCertificatesInput, ...func(*acm.Options)) (*acm.ListCertificatesOutput, error) {
+			return &acm.ListCertificatesOutput{CertificateSummaryList: summaries}, nil
+		},
+		describe: func(ctx context.Context, in *acm.DescribeCertificateInput, _ ...func(*acm.Options)) (*acm.DescribeCertificateOutput, error) {
+			current := active.Add(1)
+			for current > peak.Load() && !peak.CompareAndSwap(peak.Load(), current) {
+			}
```
[the remark this answers is not in the document]
Comment (YoungJinJung) [reply internal/services/aws/acm_test.go]: Fixed in f40c166. The CAS loop now reuses one observed peak value per attempt and reloads only after a failed swap, so a lower concurrent observation cannot replace a newer maximum. Validation passed: focused concurrency test x100, make test, make build, and git diff --check.

**Review by YoungJinJung — COMMENTED** [review]
Reviewed head: f40c166cba2dbfd2811cd99d20f295b7df1815dd

No actionable findings remain. I read the complete diff, issue scope, tests, CI, prior reviews/comments, and current review threads. The latest commits satisfy the canonical documentation requirement and fix the concurrency-test CAS without changing production behavior. The remaining Amazon Q range-variable warning is a false positive under this module’s Go 1.25 toolchain.

Validation: `make test`, `make build`, and `git diff --check origin/main...HEAD` pass. Required CI is green. I am not self-approving or merging because PR #297 is authored by the active YoungJinJung identity and branch protection requires an independent approval.

**Review by YoungJinJung — COMMENTED** [review]
Reviewed head f40c166cba2dbfd2811cd99d20f295b7df1815dd after the documentation and concurrency-test fixes. I read the aggregate diff, prior review threads, new commits, and current CI; no new actionable findings remain. Exact-head local validation: internal/app, internal/config, internal/domain, internal/inspector, internal/services/aws, and other non-auth packages passed; make build passed. The full local make test run was limited only by sandbox denial of the existing httptest listener in internal/auth; GitHub CI is green.

**Review by YoungJinJung — COMMENTED** [review]
Reviewed head: f40c166cba2dbfd2811cd99d20f295b7df1815dd

No actionable findings. The implementation matches #283, prior findings are resolved, CI is green, and independent local validation passed with `make test`, `make build`, and `git diff --check`.

**Review by YoungJinJung — COMMENTED** [review]
Re-reviewed head: f40c166cba2dbfd2811cd99d20f295b7df1815dd

No actionable findings. I read the complete diff, tests, CI, prior reviews/comments, and the three commits since the last substantive full-head review. Expired certificates now use unambiguous wording while certificates expiring at the evaluation instant retain the 0-day wording; the canonical English/Korean documentation covers the ACM browser, repository integration, Inspector rule, and configurable threshold; and the concurrency peak test now uses a stable compare-and-swap loop.

Exact-head validation passes with `GOCACHE=/private/tmp/unic-go-cache make test`, `GOCACHE=/private/tmp/unic-go-cache make build`, `GOCACHE=/private/tmp/unic-go-cache go test -race ./internal/services/aws ./internal/inspector`, and `git diff --check origin/main...HEAD`. Required CI and CodeRabbit checks are green. I am not self-approving or merging because this PR is authored by the active YoungJinJung identity and branch protection requires an independent approval.

**Review by YoungJinJung — COMMENTED** [review]
Reviewed head f40c166cba2dbfd2811cd99d20f295b7df1815dd. 이전 P2는 해결됐습니다: 만료 인증서와 오늘 만료 인증서를 분리해 표현하고 회귀 테스트를 추가했습니다. 문서 보완과 concurrency CAS 수정까지 증분 diff를 확인했으며 새 actionable finding은 없습니다.
